clients/registry-amqp/src/main/java/org/eclipse/hono/client/registry/amqp/ProtonBasedCredentialsClient.java

```diff
@@ -51,24 +56,32 @@
 /**
  * A vertx-proton based client of Hono's Credentials service.
  *
+ * If a response cache has been provided, a notification receiver can be used to receive notifications about changes in
+ * tenants, device registrations and credentials from Hono's Device Registry. The notifications are used to invalidate
+ * the corresponding entries in the response cache.
  */
 public class ProtonBasedCredentialsClient extends AbstractRequestResponseServiceClient<CredentialsObject, CredentialsResult<CredentialsObject>> implements CredentialsClient {
 
     private static final Logger LOG = LoggerFactory.getLogger(ProtonBasedCredentialsClient.class);
     private static final String TAG_AUTH_ID = "auth_id";
     private static final String TAG_CREDENTIALS_TYPE = "credentials_type";
+    private static final String ATTRIBUTE_KEY_DEVICE_ID = "device-id";
+
+    private final NotificationReceiver notificationReceiver;
 
     /**
      * Creates a new client for a connection.
      *
      * @param connection The connection to the Credentials service.
      * @param samplerFactory The factory for creating samplers for tracing AMQP messages being sent.
+     * @param notificationReceiver The client to receive notifications from the device registry.
```

Comment:
please add some text describing that the given receiver will be started/stopped as part of this component's lifecycle